src/app/game-state/log.service.ts

```diff
@@ -49,20 +50,36 @@ export class LogService {
       log = this.craftingLog;
     }
 
-    let newMessage = {
+    let newMessage: Log = {
       message: message,
       type: type,
       topic: topic,
       timestamp: Date.now()
     };
-    log.unshift(newMessage);
+
+    if (log.length == 0 || ((newMessage.timestamp - log[0].timestamp) > LOG_MERGE_INTERVAL_MS) || !log[0].message.includes(newMessage.message)) {
+      // Initialization || Repeat Not Found || Repeat is not within 1 second
+      log.unshift(newMessage);
+      if (this.logTopics.includes(topic)) {
+        this.addToCurrentLog(newMessage);
+      }
```

Comment:
Moved `addToCurrentLog` here as we do not want to call the function if it is a merge.